src/rockset_sqlalchemy/client/cursor.py

```diff
@@ -74,6 +78,21 @@ def fetchone(self):
 
         return tuple(result)
 
+    def _response_to_column_fields(self, column_fields):
+        if column_fields:
+            columns = [cf["name"] for cf in column_fields]
+
+        schema = rockset.Document()
+        for r in self._get_docs(self._response):
+            schema.update(r)
+
+        return schema.fields(columns=columns)
+
+    def _get_docs(self, response):
+        if not response.results:
+            return []
+        return [rockset.Document(row) for row in response.results]
```

Comment:
instead of iterating thru all the records of the result set, is it enough to just look at the very first record of the result set? (sqlalchemy is typically used for relational tables with no sparse fields)